File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/intercept/InterceptAdapter.java
package com.adadapted.android.sdk.core.intercept;

import com.adadapted.android.sdk.core.session.Session;

import java.util.Set;

public interface InterceptAdapter {
    void retrieve(final Session session, Callback callback);
    void sendEvents(final Session session, final Set<InterceptEvent> events);


    interface Callback {
        void onSuccess(Intercept intercept);
    }
}

File: advertising_sdk/src/main/java/com/adadapted/android/sdk/core/intercept/InterceptAdapter.kt
package com.adadapted.android.sdk.core.intercept

import com.adadapted.android.sdk.core.session.Session

interface InterceptAdapter {
    interface Callback {
        fun onSuccess(intercept: Intercept)
    }
    fun retrieve(session: Session, callback: Callback)
    fun sendEvents(session: Session, events: MutableSet<InterceptEvent>)
}